### PR TITLE
Automating newsletter issue

### PR DESCRIPTION
**Pull request by digitronik**
This month issue we forgot to create :) so decided to automate this.

- It will create newsletter collection issue At 00:30 on day-of-month 1.
- It will add notification to IRC/Telegram about issue open

example:
- https://github.com/digitronik/meetup-talks/issues/2
- https://github.com/digitronik/meetup-talks/actions/runs/181304681

part of #15

### CODE REVIEW

**Comment by digitronik** [comment]
@Pradhvan can you please review this?... as per my info this month you pointed this issue.

File: .github/ISSUE_TEMPLATE/newsletter_box.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+name: Newsletter
```
Comment (bhavin192) [inline .github/ISSUE_TEMPLATE/newsletter_box.md:2]: Let's call it 'Newsletter collection box'?

File: .github/ISSUE_TEMPLATE/newsletter_box.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+title: "{{ env.EVENT }} PythonPune Newsletter Collection Box"
```
Comment (bhavin192) [inline .github/ISSUE_TEMPLATE/newsletter_box.md:3]:
```suggestion
title: "{{ env.EVENT }} PythonPune newsletter collection box"
```

File: .github/ISSUE_TEMPLATE/newsletter_box.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+labels: newsletter
+---
+
+Let's use this issue to collect python related news, new package updates and events happening around Pune.
```
Comment (bhavin192) [inline .github/ISSUE_TEMPLATE/newsletter_box.md:7]:
```suggestion
Let's use this issue to collect Python related news, new package updates and events happening around Pune.
```

File: .github/workflows/newsletter-issue.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  issue:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Event edition
+        id: event
+        run: echo "::set-output name=edition::$(date +'%B-%Y')"
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:15]:
```suggestion
      - name: Newsletter edition
        id: edition
        run: echo "::set-output name=date::$(date +'%B %Y')"
```
How about this?

File: .github/workflows/newsletter-issue.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          EVENT: ${{ steps.event.outputs.edition }}
+        with:
+          filename: .github/ISSUE_TEMPLATE/newsletter_box.md
+        id: newsletter-issue
+
+      - name: IRC notification
+        uses: Gottox/irc-message-action@v1
+        with:
+          channel: '#pythonpune'
+          nickname: gh_action
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:30]: Let's have some name like this: `pypune_${{ github.run_id }}`?

File: .github/workflows/newsletter-issue.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+          message: |-
+            ${{ steps.event.outputs.edition }} PythonPune Newsletter Collection Box issue created.
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:32]:
```suggestion
            ${{ steps.event.outputs.edition }} PythonPune newsletter collection box issue created.
```

File: .github/workflows/newsletter-issue.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+name: Newsletter Issue
+
+on:
+  schedule:
+  - cron: "30 0 1 * *"    # At 00:30 on day-of-month 1.
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:5]:
```suggestion
    - cron: "30 0 1 * *"    # At 00:30 on day-of-month 1.
```
Just for consistency :smile:

File: .github/ISSUE_TEMPLATE/newsletter_box.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+We can compile and publish it during **{{ env.EVENT }} PythonPune** meetup. 
+
+### How to add news:
+Comment on this issue with news link and news topic and also include the source from we got this.
```
Comment (bhavin192) [inline .github/ISSUE_TEMPLATE/newsletter_box.md:11]:
```suggestion
Comment on this issue with news link. Please try to include the source like tweet / reddit and a TL;DR section giving brief about the link.
```

File: .github/workflows/newsletter-issue.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Event edition
+        id: event
+        run: echo "::set-output name=edition::$(date +'%B-%Y')"
+
+      - name: Create issue
+        uses: JasonEtco/create-an-issue@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          EVENT: ${{ steps.event.outputs.edition }}
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:21]:
```suggestion
          EVENT: ${{ steps.edition.outputs.date }}
```

File: .github/workflows/newsletter-issue.yml (hunk 4 above)
Comment (digitronik) [reply .github/workflows/newsletter-issue.yml:15]: Its more general term we are started using like even and edition (Personally don't want to separate Newsletter). I would like go with `event` and its `edition`.
like PythonPune July 2020 edition.
PythonPune meet-up is our `event` and `July 2020` is our edition.

Yup I will remove `-` as we are not using it for consistency.

Comment (bhavin192) [reply .github/workflows/newsletter-issue.yml:15]: Okay, fair enough.

File: .github/ISSUE_TEMPLATE/newsletter_box.md (hunk 1 above)
Comment (digitronik) [reply .github/ISSUE_TEMPLATE/newsletter_box.md:2]: sure

File: .github/workflows/newsletter-issue.yml (hunk 5 above)
Comment (digitronik) [reply .github/workflows/newsletter-issue.yml:30]: hummm yup `pypune_gh_<run_id>` ^^

**Comment by digitronik** [comment]
@bhavin192 take a look... I tested with push event working fine.

File: .github/workflows/newsletter-issue.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          filename: .github/ISSUE_TEMPLATE/newsletter_box.md
+        id: newsletter-issue
+
+      - name: IRC notification
+        uses: Gottox/irc-message-action@v1
+        with:
+          channel: '#pythonpune_test'
```
Comment (bhavin192) [inline .github/workflows/newsletter-issue.yml:30]: Please change this.

Comment (digitronik) [reply .github/workflows/newsletter-issue.yml]: ahhh thanks, I was added that for testing.

**Comment by akshay196** [comment]
@digitronik @bhavin192 @chkumar246  Thank you for automating newsletter issue and notification. This is working great.

**Pull request merged**